### ISSUE TITLE
`typing.Literal` is not detected when renamed at import time

### ISSUE THREAD

**Issue by ntninja** [PyCQA/pyflakes#561]
I just encountered a false-positive regression reported by PyFlakes 2.2.0, but not 2.1.1: If the *Literal* class from `typing` or `typing_extensions` is renamed at import time, PyFlakes looses track of it and treats any string argument as forward-reference that then cannot be resolved – causing it to emit a *F821 undefined name 'none'* error through flake8.

Minimum example:

```py
import typing as ty
from typing_extensions import Literal as ty_Literal
		
@ty.overload
def request(
		decoder: ty_Literal["none"] = "none",
) -> None:
	...
```

**Comment by ntninja** [comment]
Actually, it doesn't detect `ty.overload` as `typing.overload` in the example above either and emits *redefinition of unused 'request' from line …* when simply duplicating the `request` function in the example.

**Comment by asottile** [comment]
I'd suggest not doing that and using `typing` and `typing_extensions` as per convention

**Comment by Akuli** [comment]
someone reported the `typing.overload` problem to mypy's issue tracker a while ago: https://github.com/python/mypy/issues/9329

>I'd suggest not doing that and ...

This seems to conflict with pyflakes README, which says that it tries to avoid false positives **regardless of style**.

**Comment by asottile** [comment]
> 
> 
> someone reported the `typing.overload` problem to mypy's issue tracker a while ago: [python/mypy#9329](https://github.com/python/mypy/issues/9329)
> 
> > I'd suggest not doing that and ...
> 
> This seems to conflict with pyflakes README, which says that it tries to avoid false positives **regardless of style.

feel free to attempt a patch

the README is actually referring to issues of style like whitespace and newlines which are handled by pycodestyle

and again the README says "tries", there are cases which are impossible to determine statically (such as when pyflakes would need to follow imports to determine if you're actually using the typing module) -- pyflakes does not do any dynamic analysis

**Comment by davidism** [comment]
> I'd suggest not doing that

I do `import typing as t` because the other two options are `import typing; typing.xyz` which gets very verbose, or `from typing import xyz` which pollutes the namespace with lots of very generic names.

I can work around this by *also* doing `import typing; typing.overload` or `from typing import overload`, but it feels like that goes against the spirit of linting.

I get why this is hard to detect given pyflake's design. This has probably already been considered, but does Python's AST not associate the original target with the alias so it can be introspected? Is that something that should be reported to Python?

**Comment by asottile** [comment]
>  but does Python's AST not associate the original target with the alias so it can be introspected? Is that something that should be reported to Python?

it does not, and should not be reported, the ast doesn't give you any insight into how names will be bound at runtime

pyflakes has some code here to determine that (with scopes, etc.), the patch to fix this issue is probably pretty simple -- but nobody has stepped up to attempt it.

**Comment by davidism** [comment]
For someone who can dive into this: it looks like [`ast.Import` / `ImportFrom` nodes](https://docs.python.org/3/library/ast.html#imports) have the module name and imported names, and [pyflakes is already doing something](https://github.com/PyCQA/pyflakes/blob/26cf0631fe89f61d5b0ef8d6949676f051e35796/pyflakes/checker.py#L2183) with that. Not sure what it would need to change to keep track of `t.overload = typing.overload`, but that's probably a good place to start.

**Comment by angus-lherrou** [comment]
I believe I've solved this via #632.
